posts/2022-08-02-separate-stack-trace-22008.adoc

```diff
@@ -0,0 +1,466 @@
+|===
+
+For a list of past security vulnerability fixes, reference the link:{url-prefix}/docs/latest/security-vulnerabilities.html[Security vulnerability (CVE) list].
+
+
+[#bugs]
+== Notable bugs fixed in this release
+
+We’ve spent some time fixing bugs. The following sections describe just some of the issues resolved in this release. If you’re interested, here’s the  link:https://github.com/OpenLiberty/open-liberty/issues?q=label%3Arelease%3A22008+label%3A%22release+bug%22[full list of bugs fixed in 22.0.0.8].
+
+* link:https://github.com/OpenLiberty/open-liberty/issues/21526[UI generated by openapi-3.1 feature doesn't show the link specific endpoints]
++
+An issue arose when you clicked an endpoint in the UI that is generated by the `openapi-3.1 feature`, the clicked endpoint is expanded however the URL of the browser is not changed. As the result, you could not pass a link to a specific endpoint to others for reference. The URL of the browser is expected to change to show the link to the clicked endpoint.
++
+This was not a problem and used to work in the UI that is generated by the `apiDiscovery-1.0` feature and also works in the `petstore` demo UI that can be found link:https://petstore.swagger.io/#/pet/addPet[here].
++
+The links below shows the issue at hand:
++ 
+link:https://user-images.githubusercontent.com/9754967/175326096-20dc320b-a6f2-4c81-a745-c71f9f7fe4b3.png[UI generated by openapi-3.1 does not show the correct URL]
++
+link:https://user-images.githubusercontent.com/9754967/175325800-51ff7066-135d-497a-91aa-f1e7a7469716.png[UI generated by apiDiscovery-1.0 shows the correct URL]
++ 
+link:https://user-images.githubusercontent.com/9754967/175325883-6bddc8bc-dc7a-4ebb-98ad-42e57dc4f36e.png[UI in petstore shows the correct URL]
++
+Diagnostic information:
++
+OpenLiberty Version: `22.0.0.4`
+Affected feature(s): `openapi-3.1`, `mpOpenApi` (all verssions)
+Java Version:
++
+[source, java]
+----
+sh-4.4$ java -version
+Picked up JAVA_TOOL_OPTIONS: -XX:+IgnoreUnrecognizedVMOptions -XX:+UseContainerSupport -XX:+IdleTuningCompactOnIdle -XX:+IdleTuningGcOnIdle
+openjdk version "11.0.15" 2022-04-19
+OpenJDK Runtime Environment Temurin-11.0.15+10 (build 11.0.15+10)
+OpenJDK 64-Bit Server VM Temurin-11.0.15+10 (build 11.0.15+10, mixed mode)
+----
++
+`server.xml` cofiguration
++
+[source,xml]
+----
+<featureManager>
+    <feature>jaxrs-2.0</feature>
+    <feature>ssl-1.0</feature>
+    <feature>openapi-3.1</feature>
+    <feature>concurrent-1.0</feature>
+    <feature>monitor-1.0</feature>
+    <feature>localConnector-1.0</feature>
+</featureManager>
+----
++ 
+This issue has since been reviewed and subsequently the affected features list have been updated once the UI is shared with the `mpOpenApi` features. This did previously work for openapi-3.1 but had been broken by link:https://github.com/OpenLiberty/open-liberty/pull/19535[#19535] in 22.0.0.2. This issue has now been resolved. 
```

Comment:
```suggestion
* link:https://github.com/OpenLiberty/open-liberty/issues/21526[UI generated by openapi-3.1 feature doesn't show the link specific endpoints]
+
An issue arose when you clicked an endpoint in the UI that is generated by the `openapi-3.1 feature`, the clicked endpoint is expanded however the URL of the browser is not changed. As the result, you could not pass a link to a specific endpoint to others for reference. The URL of the browser is expected to change to show the link to the clicked endpoint.
+
This issue has since been reviewed and subsequently the affected features list have been updated once the UI is shared with the `mpOpenApi` features. This did previously work for openapi-3.1 but had been broken by link:https://github.com/OpenLiberty/open-liberty/pull/19535[#19535] in 22.0.0.2. This issue has now been resolved. 
```